providers/gce/gce_loadbalancer_internal_test.go

```diff
@@ -705,6 +705,32 @@ func TestUpdateInternalLoadBalancerNodes(t *testing.T) {
 	)
 }
 
+func TestUpdateInternalLoadBalancerNodesWithEmptyZone(t *testing.T) {
+	t.Parallel()
+
+	vals := DefaultTestClusterValues()
+	gce, err := fakeGCECloud(vals)
+	require.NoError(t, err)
+	nodeName := "test-node-1"
+	node1Name := []string{nodeName}
+
+	svc := fakeLoadbalancerService(string(LBTypeInternal))
+	svc, err = gce.client.CoreV1().Services(svc.Namespace).Create(context.TODO(), svc, metav1.CreateOptions{})
+	require.NoError(t, err)
+	nodes, err := createAndInsertNodes(gce, node1Name, vals.ZoneName)
+	nodes[0].Labels[v1.LabelTopologyZone] = "" // empty zone
+	require.NoError(t, err)
+
+	_, err = gce.ensureInternalLoadBalancer(vals.ClusterName, vals.ClusterID, svc, nil, nodes)
+	assert.NoError(t, err)
+
+	// Expect load balancer to not have deleted node test-node-1
+	node := &v1.Node{ObjectMeta: metav1.ObjectMeta{Name: nodeName}}
+	exist, err := gce.InstanceExists(context.TODO(), node)
```

Comment:
I think this test is wrong
it is supposed to test if a node is present in the instance group and then if that node zone is removed that it won't be removed from the group

I see something like
- create fake node with a zone
- create service
- make sure the node is in the instance group
- remove zone from the node
- trigger update somehow (node modification should do it?)
- make sure node is still in the instance group

here you just check if the instance exists (and it will as long as the node exists)

Reply:
You are right, working on it